src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageServiceHost.cs

```diff
@@ -25,14 +26,11 @@ protected override Guid IntelliSenseProviderGuid
             get { return CSharpIntellisenseProvider; }
         }
 
-        /// <summary>
-        /// Invoked when the UnconfiguredProject is first loaded to initialize language services.
-        /// </summary>
-        [ProjectAutoLoad(ProjectLoadCheckpoint.InitialActiveConfigurationKnown)]
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.CSharpLanguageService)]
-        private void Initialize()
+        private Task OnProjectFactoryCompleted()
         {
-            var nowait = InitializeAsync();
+            return InitializeAsync();
```

Comment:
Don't we want to await InitializeAsync() here so that InitializeAsync completes by the time we exit this method?